src/server/editorServices.ts

```diff
@@ -1746,6 +1760,10 @@ namespace ts.server {
                 if (!openedByClient) {
                     this.watchClosedScriptInfo(info);
                 }
+                else if (!isRootedDiskPath(fileName) && currentDirectory !== this.currentDirectory) {

```

Comment:
`this.currentDirectory` on the editor service as a whole seems wrong to me.. when do you ever want the server directory to resolve a file reference... do not know what to do about it though.. I think the reason it works is that the editors always send us absolute paths. but it just seems wrong

Reply:
I agree.. But I think for now we should let it sit that way and later just get rid of this.currentDirectory from projectService and ensure all the file paths are absolute?

Reply:
Also i dont know what to do if non rooted file is opened without providing projectRootPath? which was one of the reason I let it be that way.. (what do we use as current directory then?) 
PS: I am updating the current test to run both with and without projectRootPath when opening the file. 
Have noticed that if you openFolder in vscode the projectRootPath is sent as that folder path,

Reply:
I think we should just assume that anything that comes from the editor is a `Path`. do not mess up with it. only closed files, or files that we have generated by resolution should be converted to a `Path` by us, and for these we always have a directory to resolve the path against. 

in other words, if the editor gives us a file name, we should not change it.